mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala

```diff
@@ -293,12 +298,27 @@ class NaiveBayes private (
       }
     }
 
+    val requireZeroOneBernoulliValues: Vector => Unit = (v: Vector) => {
+      if (modelType == "Bernoulli") {
+        val values = v match {
+          case SparseVector(size, indices, values) =>
+            values
+          case DenseVector(values) =>
+            values
+        }
+        if (!values.forall(v => v == 0.0 || v == 1.0)) {
+          throw new SparkException(s"Bernoulli Naive Bayes requires 0 or 1 feature values but found $v.")
+        }
+      }
+    }
+
     // Aggregates term frequencies per label.
     // TODO: Calling combineByKey and collect creates two stages, we can implement something
     // TODO: similar to reduceByKeyLocally to save one stage.
     val aggregated = data.map(p => (p.label, p.features)).combineByKey[(Long, BDV[Double])](
       createCombiner = (v: Vector) => {
         requireNonnegativeValues(v)
+        requireZeroOneBernoulliValues(v)
```

Comment:
Could you please move the ```if (modelType == "Bernoulli")``` check here so that only one of these require() methods is called?